src/app/wallets/update-on-chain-receipt.ts

```diff
@@ -179,11 +179,20 @@ const processTxForWallet = async (
             walletAddressReceiver.usdToCreditReceiver.amount,
           ) as DisplayCurrencyBaseAmount
 
+          const displayCurrency = DisplayCurrency.Usd
+
           const metadata = LedgerFacade.OnChainReceiveLedgerMetadata({
             onChainTxHash: tx.rawTx.txHash,
-            fee: walletAddressReceiver.btcBankFee,
+            paymentFlow: {
+              btcPaymentAmount: walletAddressReceiver.btcToCreditReceiver,
+              usdPaymentAmount: walletAddressReceiver.usdToCreditReceiver,
+              btcProtocolFee: walletAddressReceiver.btcBankFee,
+              usdProtocolFee: walletAddressReceiver.usdBankFee,
+            } as OnChainPaymentFlowState<S, R>,
```

Comment:
Don't like the cast in a use-case function. If it must be a cast then at least put it behind a simple function. If it works for the desired dependency graph perhaps it could even be `walletAddressReceiver.intoPaymentFlowState` or the other way around `paymentFlowStateFromReceiver(..)`. Basically an explicit conversion function (which hides the cast).

Reply:
I was able to change that type to remove the cast completely
https://github.com/GaloyMoney/galoy/pull/2157/commits/d5dcccde0bfe364c408c055d28bd550277a4ba24